### PR TITLE
Add mongodb

### PR DESCRIPTION
**Pull request by YQBird**
Add basic express and mongodb

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -1,11 +1,21 @@
-import express from 'express';
-import http from 'http';
+const express = require('express');
+const mongoose = require('mongoose');
+const bodyParser = require('body-parser');
+const routes = require('./routes/news_routes');
+const app = express();
 
-var app = express();
+mongoose.Promise = global.Promise;
```
Comment (Jackson7683) [inline server/app.js:7]: Hi, so what is the purpose of the mongoose.Promise?

File: server/package.json [hunk 2]
```diff
@@ -4,23 +4,21 @@
   "description": "This is the server app providing REST API",
   "main": "app.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1",
-    "start": "./node_modules/babel-cli/bin/babel-node.js --presets node6 app.js"
+    "test": "NODE_ENV=testing nodemon --exec 'mocha --recursive -R min'",
```
Comment (Jackson7683) [inline server/package.json:7]: do we need specify 'test' for client side and server side, e.g., client side: npm run test = ng test

Comment (YQBird) [reply server/package.json:7]: This script is only used in /server sub folder. We could add server side test script in root folder in the future.

File: server/app.js (hunk 1 above)
Comment (YQBird) [reply server/app.js:7]: This line tells mongoose to use JavaScript default Promise.

**Review by Jackson7683 — APPROVED** [review]

**Pull request merged**